src/robot/reporting/resultwriter.py

```diff
@@ -132,7 +132,9 @@ def js_result(self):
         if self._js_result is None:
             builder = JsModelBuilder(log_path=self._settings.log,
                                      split_log=self._settings.split_log,
-                                     prune_input_to_save_memory=self._prune)
+                                     prune_input_to_save_memory=self._prune,
+                                     auto_expand_keywords
+                                     =self._settings.auto_expand_keywords)
```

Comment:
1. This formatting is a bit odd. Probably would be better to do something like
```python
builder = JsModelBuilder(
    options=here,
    with_suitable=indentation
)
```

2. The `prune_input_to_save_memory` option is somewhat special and I'd keep it as the last one.

Reply:
parameter order was updated. but implementing a new class storing options would effect more JS-modules/classes. e.g. JsBuildingContext beside JsModelBuilder. This increases code complexity where avoiding 1 extra parameter should improve code readability. Thus i prefer to reject this JS-options-refactoring...